src/components/Learningpath/components/ArticleStep.tsx

```diff
@@ -0,0 +1,124 @@
+import { BaseStepProps, learningpathStepQuery, EmbedPageContent } from "./LearningpathStep";
+import config from "../../../config";
+import { GQLLearningpathStepQuery, GQLLearningpathStepQueryVariables } from "../../../graphqlTypes";
+import { Breadcrumb } from "../../../interfaces";
+import { getArticleScripts } from "../../../util/getArticleScripts";
+import { getContentType } from "../../../util/getContentType";
+import getStructuredDataFromArticle from "../../../util/getStructuredDataFromArticle";
+import { transformArticle } from "../../../util/transformArticle";
+import Article from "../../Article";
+import { CreatedBy } from "../../Article/CreatedBy";
+import { ContentPlaceholder } from "../../ContentPlaceholder";
+import { DefaultErrorMessage } from "../../DefaultErrorMessage";
+
+interface ArticleStepProps extends BaseStepProps {
+  breadcrumbItems: Breadcrumb[];
+  subjectId?: string;
+  taxId?: string;
+  articleId?: string;
+  children?: ReactNode;
+}
+
+export const ArticleStep = ({
+  learningpathStep,
+  skipToContentId,
+  subjectId,
+  breadcrumbItems,
+  articleId,
+  taxId,
+  children,
+}: ArticleStepProps) => {
+  const { t, i18n } = useTranslation();
+  const location = useLocation();
+
+  const { data, loading } = useQuery<GQLLearningpathStepQuery, GQLLearningpathStepQueryVariables>(
+    learningpathStepQuery,
+    {
+      variables: {
+        articleId: articleId ?? learningpathStep.resource?.article?.id.toString() ?? "",
+        resourceId: taxId ?? "",
+        includeResource: !!taxId,
+        transformArgs: {
+          path: location.pathname,
+          subjectId,
+          prettyUrl: true,
+        },
+      },
+      skip:
+        !!learningpathStep.resource?.article ||
+        !articleId ||
+        (!learningpathStep.embedUrl && !learningpathStep.resource),
+    },
+  );
+
+  const url = !learningpathStep.resource?.url ? data?.node?.url : undefined;
+  const contentUrl = url ? `${config.ndlaFrontendDomain}${url}` : undefined;
+
+  const [article, scripts] = useMemo(() => {
+    const article = learningpathStep.resource?.article ? learningpathStep.resource.article : data?.article;
+    if (!article) return [undefined, undefined];
+    return [
+      transformArticle(article, i18n.language, {
+        path: `${config.ndlaFrontendDomain}/article/${article.id}`,
+        subject: subjectId,
+        articleLanguage: article.language,
+        contentType: getContentType(learningpathStep.resource),
+      }),
+      getArticleScripts(article, i18n.language),
+    ];
+  }, [data?.article, i18n.language, learningpathStep.resource, subjectId]);
+
+  if (loading) {
+    return (
+      <EmbedPageContent>
+        <ContentPlaceholder variant="article" />
+      </EmbedPageContent>
+    );
+  }
+
+  if (!article || !scripts) {
+    return null;
+  }
+
+  const learningpathStepResource = learningpathStep.resource ?? data;
+  const resource = learningpathStep.resource ?? data?.node;
```

Comment:
Ser det er en del håndtering av at learningPathStep.resource kan være undefined i denne filen, men det kan vel fjernes ettersom du allerede sjekker for undefined i `LearningpathStep`? eller er det stress å få til med gql typene?

Reply:
Det er litt stress samt. dette er egentlig bare kopiert over fra den svære `LearningpathEmbed` fila. Litt redd for å tulle med noe 🤠